test/jasmine/tests/image_test.js

```diff
@@ -669,5 +667,38 @@ describe('image hover:', function() {
                 .then(done);
             });
         });
+
+        [
+            [true, true],
+            [true, 'reversed'],  // the default image layout
+            ['reversed', true],
+            ['reversed', 'reversed']
+        ].forEach(function(test) {
+            it('should show correct hover info regardless of axis directions ' + test, function(done) {
+                var mockCopy = Lib.extendDeep({}, mock);
+                mockCopy.layout.xaxis.autorange = test[0];
+                mockCopy.layout.yaxis.autorange = test[1];
+                mockCopy.data[0].colormodel = 'rgba';
+                mockCopy.data[0].hovertemplate = 'x:%{x}, y:%{y}, z:%{z}<extra></extra>';
+                Plotly.newPlot(gd, mockCopy)
+                .then(function() {
+                    var x = 205;
+                    var y = 125;
+
+                    // adjust considering css
+                    if(test[0] === 'reversed') x = 512 - x;
+                    if(test[1] !== 'reversed') y = 512 - y;
+                    _hover(x, y);
```

Comment:
It looks like I was misunderstanding the way `_hover()` worked. Frankly, I'm still a little confused. They are neither "scene coordinates" nor screen coordinates. We are not cheating the test here, are we? ;)

Reply:
No. That's why I added `x` and `y` to the `hovertemplate` test.

Reply:
Yes, that bit is re-assuring :)